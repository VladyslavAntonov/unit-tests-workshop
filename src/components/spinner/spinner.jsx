import * as styles from './spinner.module.scss';

export function Spinner() {
  return (
    <div className={styles.spinner}>
      {Array.from(
          { length: 12 },
          (_, i) => <div className={styles['spinner__element']} key={i}/>,
      )}
    </div>
  );
}
