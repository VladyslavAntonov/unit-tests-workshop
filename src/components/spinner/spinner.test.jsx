import { shallow } from 'enzyme';

import { Spinner } from './spinner';

describe('Spinner', () => {
  let component;

  beforeEach(() => {
    component = shallow(<Spinner />);
  });

  it('matches the snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
