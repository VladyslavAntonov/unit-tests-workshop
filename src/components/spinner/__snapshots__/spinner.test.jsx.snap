// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner matches the snapshot 1`] = `
<div>
  <div
    key="0"
  />
  <div
    key="1"
  />
  <div
    key="2"
  />
  <div
    key="3"
  />
  <div
    key="4"
  />
  <div
    key="5"
  />
  <div
    key="6"
  />
  <div
    key="7"
  />
  <div
    key="8"
  />
  <div
    key="9"
  />
  <div
    key="10"
  />
  <div
    key="11"
  />
</div>
`;
